Basic/ClientDriven/Assets/Scripts/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = System.Random;
+
+namespace Unity.Multiplayer.Samples.ClientDriven
+{
+    /// <summary>
+    /// Spawn a NetworkObject at this transform's position when NetworkManer's server is started.
+    /// </summary>
+    /// <remarks>
+    /// A NetworkManager is expected to be part of the scene that this NetworkObject is a part of.
+    /// </remarks>
+    public class NetworkObjectSpawner : MonoBehaviour
+    {
+        [SerializeField]
+        private NetworkObject prefabReference;
+        [SerializeField]
+        private NetworkManager m_NetworkManager;
+
+        void Start()
+        {
+            Debug.Assert(m_NetworkManager != null, "The NetworkManager needs to be referenced!");
+            if (m_NetworkManager == null)
+            {
+                return;
+            }
```

Comment:
I see a few issues here, the old approach you had of looking for the NetworkManager at runtime was better.

1. You don't need the assert
2. If you return, this script is useless. You should instead wait for the NetworkManager's singleton to be instanciated (with a coroutine), and then do what it should do.
3. You're now dependant on a reference to the networkManager's instance, meaning that you probably have to tamper a scene to assign this reference. This can lead to conflicts during pull requests.